src/test/java/org/minions/devfund/sergio/SquarelotronTest.java

```diff
@@ -0,0 +1,347 @@
+                {16, 17, 18, 19, 20}, {21, 22, 23, 24, 25}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected),
+                Squarelotron.showSquare(squarelotron.getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 4x4.
+     */
+    @Test
+    public void testGetNumberOfRingsInA4x4Square() {
+        final int size = 4;
+        squarelotron = new Squarelotron(size);
+        final int ringsExpected = 2;
+        assertEquals(ringsExpected, squarelotron.getNumberOfRings());
+    }
+
+    /**
+     * Tests the values in the square of 5x5.
+     */
+    @Test
+    public void testGetNumberOfRingsInA5x5Square() {
+        final int size = 5;
+        squarelotron = new Squarelotron(size);
+        final int ringsExpected = 3;
+        assertEquals(ringsExpected, squarelotron.getNumberOfRings());
+    }
+
+    /**
+     * Tests the values in the square of 3x3.
+     */
+    @Test
+    public void testRingCheckTrueInA3x3Square() {
+        final int size = 3;
+        squarelotron = new Squarelotron(size);
+        assertTrue(squarelotron.ringCheck(1, 1, 2));
+    }
+
+    /**
+     * Tests the values in the square of 4x4.
+     */
+    @Test
+    public void testRingCheckFalseInA4x4Square() {
+        final int size = 4;
+        squarelotron = new Squarelotron(size);
+        assertFalse(squarelotron.ringCheck(0, 0, 2));
+    }
+
+    /**
+     * Tests the values in the square of 5x5.
+     */
+    @Test
+    public void testUpsideDownFlipInA5x5SquareRingOne() {
+        final int size = 5;
+        final int[][] expected = {{21, 22, 23, 24, 25}, {16, 7, 8, 9, 20}, {11, 12, 13, 14, 15},
+                {6, 17, 18, 19, 10}, {1, 2, 3, 4, 5}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.upsideDownFlip(1)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 6x6.
+     */
+    @Test
+    public void testUpsideDownFlipInA6x6SquareRingTwo() {
+        final int size = 6;
+        final int[][] expected = {{1, 2, 3, 4, 5, 6}, {7, 26, 27, 28, 29, 12}, {13, 20, 15, 16, 23, 18},
+                {19, 14, 21, 22, 17, 24}, {25, 8, 9, 10, 11, 30}, {31, 32, 33, 34, 35, 36}};
+        squarelotron = new Squarelotron(size);
+        assertEquals(Squarelotron.showSquare(expected), Squarelotron.showSquare(squarelotron.upsideDownFlip(2)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 6x6.
+     */
+    @Test
+    public void testUpsideDownFlipIn6x6SquareRingThree() {
+        final int size = 6;
+        final int ring = 3;
+        final int[][] expected = {{1, 2, 3, 4, 5, 6}, {7, 8, 9, 10, 11, 12}, {13, 14, 21, 22, 17, 18},
+                {19, 20, 15, 16, 23, 24}, {25, 26, 27, 28, 29, 30}, {31, 32, 33, 34, 35, 36}};
+        squarelotron = new Squarelotron(size);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)